### ISSUE TITLE
XML catalog file with specific DOCTYPE did not seem to be loaded by the resolver

### ISSUE THREAD

**Issue by raducoravu** [xmlresolver/xmlresolver#115]
We had this XML catalog referenced in a command line using Saxon to process an XML with an XSLT which had certain imports:

        <?xml version="1.0" encoding="UTF-8"?>
        <!DOCTYPE catalog PUBLIC "-//OASIS//DTD XML Catalogs V1.1//EN" "http://www.oasis-open.org/committees/entity/release/1.1/catalog.dtd">
        <catalog xmlns="urn:oasis:names:tc:entity:xmlns:xml:catalog">
            <uri name="http://this.name.should.be.resolved.by.catalog" uri="test.xsd"/>
            <system systemId="http://this.name.should.be.resolved.by.catalog" uri="test.xsd"/>
            
            
            <uri name="http://fake.com/commons.xsl" uri="commons.xsl"/>
            
        </catalog>

and somehow the mappings were not taken into account unless we removed the DOCTYPE declaration.

**Comment by ndw** [comment]
That is very odd. I was sure there were tests for catalog files with doctype declarations, but perhaps I'm mistaken. I'll look for a fix this week.

**Comment by ndw** [comment]
I can't reproduce this. I created a test with a catalog that used the doctype declaration you suggested. It resolves without any problem, AFAICT.

You can get the test from the `iss115` branch at `https://github.com/ndw/xmlresolver/`

Can you provide any more details about how to reproduce this?

**Comment by AlexJitianu** [comment]
Hi,

I looked at the test and sample catalog, but I didn't notice a possible cause why it doesn't catch the situation. On our side, we reproduced the issue with Saxon 11, so I'm attaching the samples.
I run the command:
`java -cp saxon-he-11.4.jar;lib\xmlresolver-4.4.3.jar net.sf.saxon.Transform -xsl:catalog-test/main.xsl -catalog:catalog-test/catalog-with-dtd.xml -it:start`

I notice how the process lingers for awhile, it might be trying to go on the web to retrieve `catalog.dtd`, but it doesn't have the proxy setup. So I get this:
```
Error on line 7 column 50 of main.xsl:
  XTSE0165  I/O error reported by XML parser processing urn:modules:include.xsl. Caused by
  java.net.MalformedURLException: unknown protocol: urn
I/O error reported by XML parser processing urn:modules:include.xsl
```
Just by removing the DOCTYPE definition from the catalog, it all begins to work.
I've made the test with version 4.5.1 and it yields the same result.

[catalog-test.zip](https://github.com/xmlresolver/xmlresolver/files/9596399/catalog-test.zip)
